lib/postmessage/postmessage.js

```diff
@@ -0,0 +1,438 @@
+          hash.source && hash.source.name != null && hash.source.url && hash.postmessage)) {
+          // ignore since hash could've come from somewhere else
+          return;
+        }
+      }
+      catch (ex) {
+        // ignore since hash could've come from somewhere else
+        return;
+      }
+      var msg = hash.postmessage,
+        cbs = pm.data("callbacks.postmessage") || {},
+        cb = cbs[msg.type];
+      if (cb) {
+        cb(msg.data);
+      }
+      else {
+        var source_window;
+        if (hash.source.name === "parent") {
+          source_window = window.parent;
+        }
+        else {
+          source_window = window.frames[hash.source.name];
+        }
+        var l = pm.data("listeners.postmessage") || {};
+        var fns = l[msg.type] || [];
+        for (var i=0,len=fns.length; i<len; i++) {
+          var o = fns[i];
+          if (o.origin) {
+            var origin = /https?\:\/\/[^\/]*/.exec(hash.source.url)[0];
+            if (o.origin !== '*' && origin !== o.origin) {
+              console.warn("postmessage message origin mismatch", origin, o.origin);
+              if (msg.errback) {
+                // notify post message errback
+                var error = {
+                  message: "postmessage origin mismatch",
+                  origin: [origin, o.origin]
+                };
+                pm.send({target:source_window, data:error, type:msg.errback, hash:true, url:hash.source.url});
+              }
+              continue;
+            }
+          }
+
+          function sendReply ( data ) {
+            if (msg.callback) {
+              pm.send({target:source_window, data:data, type:msg.callback, hash:true, url:hash.source.url});
+            }
+          }
+
+          try {
+            if ( o.callback ) {
+              o.fn(msg.data, sendReply);
+            } else {
+              sendReply ( o.fn(msg.data) );
+            }
+          }
+          catch (ex) {
+            if (msg.errback) {
+              // notify post message errback
+              pm.send({target:source_window, data:ex, type:msg.errback, hash:true, url:hash.source.url});
+            } else {
+              throw ex;
+            }
+          }
+        };
+      }
+    },
+
+    _url: function(url) {
+      // url minus hash part
+      return (""+url).replace(/#.*$/, "");
+    }
+
+  };
+
+  $.extend(pm, {
+    defaults: {
+      target: null,  /* target window (required) */
+      url: null,     /* target window url (required if no window.postMessage or hash == true) */
+      type: null,    /* message type (required) */
+      data: null,    /* message data (required) */
+      success: null, /* success callback (optional) */
+      error: null,   /* error callback (optional) */
+      origin: "*",   /* postmessage origin (optional) */
+      hash: false    /* use location hash for message passing (optional) */
+    }
+  });
+
+})(window, typeof jQuery === "undefined" ? NO_JQUERY : jQuery);
```

Comment:
This is the main line that got changed from postmessage.js. Previously it was
 })(this, typeof jQuery === "undefined" ? NO_JQUERY : jQuery);